storage.go

```diff
@@ -4,229 +4,156 @@ import (
 	"context"
 	"encoding/json"
 	"strings"
-	"time"
 
 	"github.com/cenkalti/backoff"
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/micrologger"
 	"github.com/giantswarm/microstorage"
-	"github.com/giantswarm/operatorkit/tpr"
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 	"k8s.io/apimachinery/pkg/api/errors"
-	apismeta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apismetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes"
 	api "k8s.io/client-go/pkg/api/v1"
 )
 
-const (
-	tpoCreateMaxElapsedTime = 30 * time.Second
-)
-
-type CRDConfig struct {
-	Name, Version, Description string
-}
-
-type TPOConfig struct {
-	Name, Namespace string
-}
-
 type Config struct {
-	// Dependencies.
-
+	CRDClient *k8scrdclient.CRDClient
+	G8sClient versioned.Interface
 	K8sClient kubernetes.Interface
 	Logger    micrologger.Logger
 
-	// Settings.
-
-	// CRD is the third party resource where data objects are stored.
-	CRD CRDConfig
-
-	// TPOName is the third party object used to store data. This object
-	// will be created inside a third party resource specified by CRD. If
-	// the object already exists it will be reused. It is safe to run
-	// multiple Storage instances using the same TPO.
-	TPO TPOConfig
+	CRD       *apiextensionsv1beta1.CustomResourceDefinition
+	Name      string
+	Namespace *api.Namespace
 }
 
 func DefaultConfig() Config {
 	return Config{
-		// Dependencies.
-
-		K8sClient: nil, // Required.
-		Logger:    nil, // Required.
-
-		// Settings.
-
-		CRD: CRDConfig{
-			Name:        "crd-storage.giantswarm.io",
-			Version:     "v1",
-			Description: "Storage data managed by github.com/giantswarm/crdstorage",
-		},
-
-		TPO: TPOConfig{
-			Name:      "", // Required.
-			Namespace: "giantswarm",
-		},
+		CRDClient: nil,
+		G8sClient: nil,
+		K8sClient: nil,
+		Logger:    nil,
+
+		CRD:       nil,
```

Comment:
Can we default to the one from apiextensions?

Reply:
Done.